### ISSUE TITLE
Check against ban list

### ISSUE THREAD

**Issue by marumari** [marumari/decklist#11]
For Standard, Modern, or Legacy, verify cards against the ban list.

**Comment by ChaimW** [comment]
I'm probably going to work on this next.

**Comment by Nightfirecat** [comment]
For Standard, you may want to look into using <whatsinstandard.com> to help with set verification logic.

**Comment by tooomm** [comment]
Where do you get your card info from? @april 

In case you use https://github.com/mtgjson/mtgjson to create your minimized data, they might have data on banned cards for several tournament types, too.

All changes are posted to `Banned and Restricted Announcement` articles in the official News on magic.wiazrds.com on a monthly basis.
Official up-to-date lists from wizards are always available and could be scraped from here:
https://magic.wizards.com/en/game-info/gameplay/rules-and-formats/banned-restricted

**Comment by ChaimW** [comment]
a good portion of the necessary information is there, it is just not checked @tooomm

**Comment by april** [comment]
Yes, it's all pulled from mtgjson. Note that the site is designed to work completely offline, so it includes an entire copy of the card database in the JavaScript, for example:

"spikeshot elder":{c:"D",b:"s",m:1,t:"2",n:"Spikeshot Elder"}

Could probably add a ban key, such as:

`b:"sm"` (aka banned in Standard and Modern)

**Comment by tooomm** [comment]
You use `b:` already, but no idea what it stands for, haha. [That file](https://github.com/april/decklist/blob/master/js/cards/decklist-cards-min.js) and it's values is a bit cryptic to be honest. Besides `m` for cmc and `n` for name I have no real clue. Also, why do you double the name information?
Using mtgjson for legality info should be the easiest implementation then because you use it already.

If possible I suggest you get the information from wizards directly at one point, if one can parse that page properly...

I guess you use some kind of script to generate your database? Is that part of the repo?

**Comment by Nightfirecat** [comment]
[generatecards.sh](https://github.com/april/decklist/blob/master/js/cards/generatecards.sh) does the work of updating the cards definition file.

**Comment by april** [comment]
This is the comment in the code for what the keys mean:

```
# Just FYI!
# b (banned) = [sml] (standard, modern, legacy)
# c (color) = White = A, Blue = B, Black = C, Red = D, Green = E, Gold = F, Artifact = G , Split = S, Unknown = X, Land = Z
# m (CMC) = N  (Split = 98, Land = 99)
# n (actual name) = 'true name nemesis' to 'True Name Nemesis'
# t (type) = 1 = land, 2 = creature, 3 = instant or sorcery 4 = other
```

**Comment by april** [comment]
So I guess I already have the ban list in there, it's just not exposed obviously.  :)

**Comment by tooomm** [comment]
Haha, I was about to link to that in [parsecards.py](https://github.com/april/decklist/blob/master/js/cards/parsecards.py#L5-L10) @april :D
Found it thanks to @Nightfirecat's link.

Yep, it looks like it's already included. Great!

**Comment by ChaimW** [comment]
Yeah, it's a partial ban list (no legacy yet, because of banned v. restricted), since my semester has nearly finished, I'll probably go back to working on that soon.

**Comment by ChaimW** [comment]
I was thinking v for vintage banned, r for vintage restricted.

**Comment by april** [comment]
Legacy is already in there, there's only Vintage that isn't.

**Comment by tooomm** [comment]
Why are card names printed twice in the database?

**Comment by april** [comment]
The key is what is likely to be entered into a decklist and the name is the canonical name that appears on the PDF.

**Comment by tooomm** [comment]
Are they any different? All entries I checked were identical... besides capitalization.
Or is there another reason that you generate and keep both?

**Comment by Fryyyyy** [comment]
Much like the multiple pages, I also do format legality checking in my fork - so feel free to copy that too if you like

**Comment by ChaimW** [comment]
@tooomm The idea is that the key used to locate the correct card is all lowercase, and the output is the properly capitalized form.

**Comment by ChaimW** [comment]
@Fryyyyy I'll probably do that when I get to this issue.
